README.md

```diff
@@ -1,2 +1,79 @@
-# backoff
-Exponential backoff mechanism
+(API reference)[TODO]
+
+# backoff - exponential backoff mechanism
+
+**backoff** provides an exponential backoff mechanism [1]. It enables
+to reduce contention by making a domain backs off after failing an
+operation contested by another domain, like acquiring a lock or
+performing a `CAS` operation.
+
+The more a domain consecutively failed, the longer it will back off as
+it means the operation is highly contonded. Backing off here means
+calling `Domain.cpu_relax` (or `Thread.yield` if running with a
+version of OCaml previous to 5.0.0).
+
+This mechanism has some drawbacks. First, it adds some delays: for
+example, when a domain releases a contended lock, another domain that
+has backed off after failing acquiring it, will still have to finish
+is back off loop before retrying. Second, this increases any
+unfairness: any other threads that arrives at that time or that has
+failed acquiring the lock for a lesser number of times is more likely
+to acquire it as it will wait less.
+
+## Example
+
+To illustrate how to use `backoff`, here is a small implementation of
+ `Test-and-set` spin lock [2].
```

Comment:
BTW, technically this is a "test and test-and-set" or TTAS lock, because it first does an `Atomic.get` (first test) and then does a boolean `Atomic.compare_and_set` (also know as test-and-set).  The first (logically redundant) test avoids taking ownership of the lock memory location (cache line) in case the lock is owned by someone else and the test-and-set would almost certainly fail (there is slim chance that the lock would be immediately released after the test).